src/inc/corinfo.h

```diff
@@ -2383,6 +2383,15 @@ class ICorStaticInfo
             CORINFO_CLASS_HANDLE        cls
             ) = 0;
 
+    // return the number of bytes needed by an instance of the class allocated on the heap
+    virtual unsigned getHeapClassSize(
+        CORINFO_CLASS_HANDLE        cls
+    ) = 0;
+
+    virtual BOOL classHasFinalizer(
```

Comment:
What is the usage pattern of these method going to be? Is `getHeapClassSize` going to be always called right after `classHasFinalizer` succeeded? If it is the case, the two methods can be combined into one to reduce chattiness.

Reply:
I can replace these methods with
BOOL canAllocateClassOnStack(CORINFO_CLASS_HANDLE cls, unsigned* classSize)
to reduce chattiness. To make it cheaper it can set classSize to 0 when it returns FALSE.

Reply:
I kept two methods but replaced classHasFinalizer with canAllocateOnStack.